interfaces/devdoc/srw_lock_ll_requirements.md

```diff
@@ -0,0 +1,88 @@
+# `srw_lock_ll` requirements
+
+## Overview
+
+`srw_lock_ll` is the implementation of a slim reader writer lock.
+
+## Exposed API
+
+```c
+#define SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES \
+    SRW_LOCK_LL_TRY_ACQUIRE_OK, \
+    SRW_LOCK_LL_TRY_ACQUIRE_COULD_NOT_ACQUIRE, \
+    SRW_LOCK_LL_TRY_ACQUIRE_INVALID_ARGS
+
+typedef void* SRW_LOCK_LL;
+
+MU_DEFINE_ENUM(SRW_LOCK_LL_TRY_ACQUIRE_RESULT, SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES)
+
+MOCKABLE_FUNCTION(, void, srw_lock_ll_init);
+MOCKABLE_FUNCTION(, void, srw_lock_ll_deinit, SRW_LOCK_LL*, srw_lock_ll);
+
+/*writer APIs*/
+MOCKABLE_FUNCTION(, void, srw_lock_ll_acquire_exclusive, SRW_LOCK_LL*, srw_lock_ll);
+MOCKABLE_FUNCTION(, SRW_LOCK_LL_TRY_ACQUIRE_RESULT, srw_lock_ll_try_acquire_exclusive, SRW_LOCK_LL*, srw_lock_ll);
```

Comment:
>SRW_LOCK_LL* [](http://example.com/codeflow?start=87&length=12)

all these parameters, typed as "SRW_LOCK_LL*" are sort of a "pointer to pointer" to void. Is that intended? (note the `*` in the argument)

Reply:
This is part of me trying to make too generic the type used. I was aiming for having our interface in the PAL be generic, but honestly in this case it is probably better to have everything generic except the type.

I was banking on the fact that SRWLOCK behind uses a PVOID to store its data. But after all, I changed my mind and simply did

typedef SRWLOCK SRW_LOCK_LL;

When it comes to Linux we'll deal with it.